Framework/Kernel/inc/MantidKernel/PropertyHelper.h

```diff
@@ -195,19 +195,23 @@ inline void appendValue(const std::string &strvalue, std::vector<T> &value) {
   }
 
   if (step == static_cast<T>(0))
-    throw std::runtime_error("Step size must be non-zero");
+    throw std::logic_error("Step size must be non-zero");
 
   // convert the input string into boundaries and run through a list
   const auto start = boost::lexical_cast<T>(strvalue.substr(0, pos));
   const auto stop = boost::lexical_cast<T>(strvalue.substr(pos + 1, numChar));
   if (start <= stop) {
+    if (start + step < start)
```

Comment:
Is this not `if (step < 0)`?

Reply:
You get a compiler warning for that since `unsigned int` goes through this as well.